### PR TITLE
Deprecate distutils

### PR DESCRIPTION
**Pull request by EinarArnason**
Replace:
`distutils.spawn.find_executable` with `shutil.which`
`distutils.version.StrictVersion` with `packaging.version.Version`
`distutils.util.strtobool` with `codechecker_common.util.strtobool`

as distutils is not available in Ubuntu noble.

### CODE REVIEW

**Comment by bruntib** [comment]
Thank you for the patch, this is a very useful change!
Sorry, another patch had been merged which caused some merge conflicts. If you fix those, then this PR can merged. Please, also remove the "pylint disable" and "TODO" comments above the changed lines.
Thanks again!

File: analyzer/codechecker_analyzer/analyzer_context.py [hunk 1]
```diff
@@ -11,7 +11,7 @@
 
 
 # pylint: disable=no-name-in-module
-from distutils.spawn import find_executable
+from shutil import which
```
Comment (whisperity) [inline analyzer/codechecker_analyzer/analyzer_context.py:14]: These should be sorted alphabetically, and check whether the pylint silencing of the warning still applies.

File: analyzer/codechecker_analyzer/analyzers/cppcheck/analyzer.py [hunk 2]
```diff
@@ -11,7 +11,7 @@
 
 from collections import defaultdict
 # TODO distutils will be removed in python3.12
```
Comment (whisperity) [inline analyzer/codechecker_analyzer/analyzers/cppcheck/analyzer.py:13]: (Looks like this TODO has been solved by this patch.)

File: analyzer/codechecker_analyzer/analyzers/gcc/analyzer.py [hunk 3]
```diff
@@ -9,7 +9,7 @@
 """
 from collections import defaultdict
 # TODO distutils will be removed in python3.12
```
Comment (whisperity) [inline analyzer/codechecker_analyzer/analyzers/gcc/analyzer.py:11]: Ditto.

**Review by bruntib — APPROVED** [review]

**Pull request merged**